app/controllers/oauth/applications_controller.rb

```diff
@@ -109,12 +121,49 @@ def set_user
     end
 
     def app_params
-      params.require(:doorkeeper_application).permit(
-        :name, :redirect_uri, :scopes, :email_subject_prefix, :lead_application_source,
-        :email_from_address, :confidential,
-        :can_access_private_user_data, :can_find_or_create_accounts, :can_message_users,
-        :can_skip_oauth_screen,
-      )
+      if @user.is_administrator?
+        params.require(:doorkeeper_application).permit(
+          :name, :redirect_uri, :scopes, :email_subject_prefix, :lead_application_source,
+          :email_from_address, :confidential,
+          :can_access_private_user_data, :can_find_or_create_accounts, :can_message_users,
+          :can_skip_oauth_screen, :member_ids
+        )
+      elsif @application.owner.member_ids.include?(@user.id)
+        params.require(:doorkeeper_application).permit(
+          :redirect_uri
+        )
+      end
+    end
+
+    def handle_member_ids(member_ids)
+      if valid_member_ids(member_ids)
+        if !member_ids.nil?
+          str_array = member_ids.split(" ")
+          mi_array = str_array.map(&:to_i)
+          #add the owner back to the member_ids array
+          mi_array.unshift(@application.owner.owner_ids[0])
+          @application.owner.member_ids = mi_array if @application.owner.member_ids != mi_array
```

Comment:
Why check that `mi_array` is changing the status quo?  At least here it doesn't really buy us an efficiency (may be less efficient due to the array comparison)

Reply:
Based on other coding in other frameworks, it is more efficient to do nothing with the database then to update it needlessly. Maybe the database does not get updated if there is not a change and I am not aware or Rails is super efficient with the database. The code has been changed.